pytest_parallel/__init__.py

```diff
@@ -313,9 +313,17 @@ def wait_for_responses_processor():
             process.start()
             processes.append(process)
 
-        [p.join() for p in processes]
+        exitcode = 0
+        for p in processes:
+            p.join()
+            exitcode |= p.exitcode
+
+        # Only wait for all tasks to complete if the workers returned ok.
+        # Otherwise, we have no guarantee that the tasks have been marked
+        # completed.
+        if not exitcode:
+            queue.join()
```

Comment:
Under what circumstances is this necessary? How could any further modifications ever be made to the queue if all threads have been joined anyway? It seems like this `queue.join()` can only ever be either a no-op or a deadlock.

I'm wondering whether the most correct thing to do here is to remove this line entirely.

Reply:
I don't know if it's actually possible for the process "join" to race with the dequeued task being marked as done. I wanted to preserve the logic just in the case.

Reply:
I have done some reading and couldn't find any reason for the task to be waited. I'll remove it, let's see how it goes.

Let me know if you happy, and I'll merge it.

Reply:
Yep, it's also my read that it's impossible for the queue to be modified after the workers have joined, so removing the queue join looks good to me.